pkg/jobs/registry.go

```diff
@@ -1485,6 +1485,30 @@ func TestingClearConstructors() func() {
 
 }
 
+// TestingRegisterConstructor is like RegisterConstructor but returns a cleanup function
+// resets the registration for the given type.
+func TestingRegisterConstructor(typ jobspb.Type, fn Constructor, opts ...RegisterOption) func() {
+	var cleanupFn func()
+	if constructorFn, found := globalMu.constructors[typ]; found {
+		opts := globalMu.options[typ]
+		cleanupFn = func() {
+			globalMu.Lock()
+			defer globalMu.Unlock()
+			globalMu.constructors[typ] = constructorFn
+			globalMu.options[typ] = opts
```

Comment:
nit: i think this is clean, but to prevent some future annoying closure bug, could you rename the local `opts` here as `originalOpts`.